[CRON] API Stability Failure

[API-Stability](https://github.com/seqan/seqan3/blob/master/test/api_stability/README.md) failed.

See https://github.com/seqan/seqan3/actions/runs/781937597 for more information.

<details><summary>Error 1: <code>error: ‘operator""_sam_dna16’ has not been declared in ‘seqan3’</code></summary>

```text
[ 11%] Building CXX object CMakeFiles/sam_dna16_literal_snippet.dir/alphabet/nucleotide/sam_dna16_literal.cpp.o
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp: In function ‘int main()’:
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp:6:27: error: ‘operator""_sam_dna16’ has not been declared in ‘seqan3’
    6 |     using seqan3::operator""_sam_dna16;
      |                           ^~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp:8:13: error: ‘sam_dna16_vector’ is not a member of ‘seqan3’
    8 |     seqan3::sam_dna16_vector foo{"ACgtTA"_sam_dna16};
      |             ^~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp:9:13: error: ‘sam_dna16_vector’ is not a member of ‘seqan3’
    9 |     seqan3::sam_dna16_vector bar = "ACG==A"_sam_dna16;
      |             ^~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp:10:16: error: unable to find string literal operator ‘operator""_sam_dna16’ with ‘const char [7]’, ‘long unsigned int’ arguments
   10 |     auto bax = "A=GTT!"_sam_dna16;
      |                ^~~~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp:12:29: error: ‘foo’ was not declared in this scope
   12 |     seqan3::debug_stream << foo << "\n" << bar << "\n" << bax << "\n";
      |                             ^~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16_literal.cpp:12:44: error: ‘bar’ was not declared in this scope
   12 |     seqan3::debug_stream << foo << "\n" << bar << "\n" << bax << "\n";
      |                                            ^~~
make[5]: *** [CMakeFiles/sam_dna16_literal_snippet.dir/build.make:63: CMakeFiles/sam_dna16_literal_snippet.dir/alphabet/nucleotide/sam_dna16_literal.cpp.o] Error 1
make[5]: Target 'CMakeFiles/sam_dna16_literal_snippet.dir/build' not remade because of errors.
make[4]: *** [CMakeFiles/Makefile2:2251: CMakeFiles/sam_dna16_literal_snippet.dir/all] Error 2
Scanning dependencies of target rna4_snippet
```
</details>
<details><summary>Error 2: <code>error: ‘operator""_phred68legacy’ has not been declared in ‘seqan3’</code></summary>

```text
[ 19%] Building CXX object CMakeFiles/phred68legacy_literal_snippet.dir/alphabet/quality/phred68legacy_literal.cpp.o
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp: In function ‘int main()’:
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:7:27: error: ‘operator""_phred68legacy’ has not been declared in ‘seqan3’
    7 |     using seqan3::operator""_phred68legacy;
      |                           ^~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:10:51: error: unable to find string literal operator ‘operator""_phred68legacy’ with ‘const char [5]’, ‘long unsigned int’ arguments
   10 |     std::vector<seqan3::phred68legacy> qual_vec = "###!"_phred68legacy;
      |                                                   ^~~~~~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:13:53: error: unable to find character literal operator ‘operator""_phred68legacy’ with ‘char’ argument
   13 |     std::vector<seqan3::phred68legacy> qual_vec2 = {'#'_phred68legacy, '#'_phred68legacy,
      |                                                     ^~~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:13:72: error: unable to find character literal operator ‘operator""_phred68legacy’ with ‘char’ argument
   13 |     std::vector<seqan3::phred68legacy> qual_vec2 = {'#'_phred68legacy, '#'_phred68legacy,
      |                                                                        ^~~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:14:53: error: unable to find character literal operator ‘operator""_phred68legacy’ with ‘char’ argument
   14 |                                                     '#'_phred68legacy, '!'_phred68legacy};
      |                                                     ^~~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:14:72: error: unable to find character literal operator ‘operator""_phred68legacy’ with ‘char’ argument
   14 |                                                     '#'_phred68legacy, '!'_phred68legacy};
      |                                                                        ^~~~~~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/quality/phred68legacy_literal.cpp:14:89: error: could not convert ‘{<expression error>, <expression error>, <expression error>, <expression error>}’ from ‘<brace-enclosed initializer list>’ to ‘std::vector<seqan3::phred68solexa>’
   14 |                                                     '#'_phred68legacy, '!'_phred68legacy};
      |                                                                                         ^
      |                                                                                         |
      |                                                                                         <brace-enclosed initializer list>
make[5]: *** [CMakeFiles/phred68legacy_literal_snippet.dir/build.make:63: CMakeFiles/phred68legacy_literal_snippet.dir/alphabet/quality/phred68legacy_literal.cpp.o] Error 1
make[5]: Target 'CMakeFiles/phred68legacy_literal_snippet.dir/build' not remade because of errors.
make[4]: *** [CMakeFiles/Makefile2:4434: CMakeFiles/phred68legacy_literal_snippet.dir/all] Error 2
Scanning dependencies of target alignment_matrix_column_major_range_base_snippet
```
</details>
<details><summary>Error 3: <code>error: ‘persist’ is not a member of ‘seqan3::views’</code></summary>

```text
[ 32%] Building CXX object CMakeFiles/pairwise_alignment_solution_6_snippet.dir/home/runner/work/seqan3/seqan3/seqan3-build/api_stability_snippet-prefix/src/api_stability_snippet/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp.o
../api_stability_snippet/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp: In function ‘int main()’:
../api_stability_snippet/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp:28:115: error: ‘persist’ is not a member of ‘seqan3::views’; did you mean ‘seqan3::detail::persist’?
   28 |     for (auto const & res : seqan3::align_pairwise(seqan3::views::pairwise_combine(vec), config) | seqan3::views::persist | filter_v)
      |                                                                                                                   ^~~~~~~
In file included from ../../../../seqan3/include/seqan3/alignment/pairwise/align_pairwise.hpp:30,
                 from ../api_stability_snippet/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp:4:
../../../../seqan3/include/seqan3/core/detail/persist_view.hpp:228:23: note: ‘seqan3::detail::persist’ declared here
  228 | inline auto constexpr persist = persist_fn{};
      |                       ^~~~~~~
make[5]: *** [CMakeFiles/pairwise_alignment_solution_6_snippet.dir/build.make:63: CMakeFiles/pairwise_alignment_solution_6_snippet.dir/home/runner/work/seqan3/seqan3/seqan3-build/api_stability_snippet-prefix/src/api_stability_snippet/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp.o] Error 1
make[5]: Target 'CMakeFiles/pairwise_alignment_solution_6_snippet.dir/build' not remade because of errors.
make[4]: *** [CMakeFiles/Makefile2:6580: CMakeFiles/pairwise_alignment_solution_6_snippet.dir/all] Error 2
Scanning dependencies of target floor_log2_snippet
```
</details>
<details><summary>Error 4: <code>error: ‘operator""_sam_dna16’ has not been declared in ‘seqan3’</code></summary>

```text
[ 63%] Building CXX object CMakeFiles/sam_dna16_snippet.dir/alphabet/nucleotide/sam_dna16.cpp.o
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp: In function ‘int main()’:
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp:6:27: error: ‘operator""_sam_dna16’ has not been declared in ‘seqan3’
    6 |     using seqan3::operator""_sam_dna16;
      |                           ^~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp:8:33: error: unable to find character literal operator ‘operator""_sam_dna16’ with ‘char’ argument
    8 |     seqan3::sam_dna16 my_letter{'A'_sam_dna16};
      |                                 ^~~~~~~~~~~~~
../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp:8:46: error: no matching function for call to ‘seqan3::dna16sam::dna16sam(<brace-enclosed initializer list>)’
    8 |     seqan3::sam_dna16 my_letter{'A'_sam_dna16};
      |                                              ^
In file included from ../../../../seqan3/include/seqan3/alphabet/nucleotide/dna16sam.hpp:15,
                 from ../../../../seqan3/include/seqan3/alphabet/nucleotide/sam_dna16.hpp:16,
                 from ../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp:1:
../../../../seqan3/include/seqan3/alphabet/nucleotide/nucleotide_base.hpp:52:15: note: candidate: ‘constexpr seqan3::nucleotide_base<derived_type, size>::nucleotide_base(const seqan3::nucleotide_base<derived_type, size>&) [with derived_type = seqan3::dna16sam; auto size = 16]’
   52 |     constexpr nucleotide_base(nucleotide_base const &)             noexcept = default; //!< Defaulted.
      |               ^~~~~~~~~~~~~~~
In file included from ../../../../seqan3/include/seqan3/alphabet/nucleotide/sam_dna16.hpp:16,
                 from ../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp:1:
../../../../seqan3/include/seqan3/alphabet/nucleotide/dna16sam.hpp:70:19: note:   inherited here
   70 |     using base_t::base_t;
      |                   ^~~~~~
../../../../seqan3/include/seqan3/alphabet/nucleotide/dna16sam.hpp:70:19: note:   an inherited constructor is not a candidate for initialization from an expression of the same or derived type
In file included from ../../../../seqan3/include/seqan3/alphabet/nucleotide/dna16sam.hpp:15,
                 from ../../../../seqan3/include/seqan3/alphabet/nucleotide/sam_dna16.hpp:16,
                 from ../api_stability_snippet/test/snippet/alphabet/nucleotide/sam_dna16.cpp:1:
../../../../seqan3/include/seqan3/alphabet/nucleotide/nucleotide_base.hpp:53:15: note: candidate: ‘constexpr seqan3::nucleotide_base<derived_type, size>::nucleotide_base(seqan3::nucleotide_base<derived_type, size>&&) [with derived_type = seqan3::dna16sam; auto size = 16]’
   53 |     constexpr nucleotide_base(nucleotide_base &&)                  noexcept = default; //!< Defaulted.
      |               ^~~~~~~~~~~~~~~
In file included from ../../../../seqan3/include/seqan3/alphabet/nucleotide/sam_dna16.hpp:16,
```
</details>
<details><summary>Error 5: <code>error: ‘operator""_sam_dna16’ has not been declared in ‘seqan3’</code></summary>

```text
[ 38%] Building CXX object alphabet/nucleotide/CMakeFiles/sam_dna16_test.dir/sam_dna16_test.cpp.o
../../../api_stability_unit/test/unit/alphabet/nucleotide/sam_dna16_test.cpp:16:23: error: ‘operator""_sam_dna16’ has not been declared in ‘seqan3’
   16 | using seqan3::operator""_sam_dna16;
      |                       ^~~~~~~~~~~~
In file included from ../../vendor/googletest/googletest/include/gtest/gtest.h:375,
                 from ../../../api_stability_unit/test/unit/alphabet/nucleotide/../alphabet_constexpr_test_template.hpp:8,
                 from ../../../api_stability_unit/test/unit/alphabet/nucleotide/sam_dna16_test.cpp:11:
../../../api_stability_unit/test/unit/alphabet/nucleotide/sam_dna16_test.cpp: In member function ‘virtual void sam_dna16_nucleotide_Test::TestBody()’:
../../../api_stability_unit/test/unit/alphabet/nucleotide/sam_dna16_test.cpp:33:34: error: unable to find character literal operator ‘operator""_sam_dna16’ with ‘char’ argument
   33 |     EXPECT_EQ(seqan3::complement('='_sam_dna16), 'N'_sam_dna16);
      |                                  ^~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:77:52: note: in definition of macro ‘GTEST_ASSERT_’
   77 |   if (const ::testing::AssertionResult gtest_ar = (expression)) \
      |                                                    ^~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:164:3: note: in expansion of macro ‘GTEST_PRED_FORMAT2_’
  164 |   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_NONFATAL_FAILURE_)
      |   ^~~~~~~~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest.h:2028:3: note: in expansion of macro ‘EXPECT_PRED_FORMAT2’
 2028 |   EXPECT_PRED_FORMAT2(::testing::internal::EqHelper::Compare, val1, val2)
      |   ^~~~~~~~~~~~~~~~~~~
../../../api_stability_unit/test/unit/alphabet/nucleotide/sam_dna16_test.cpp:33:5: note: in expansion of macro ‘EXPECT_EQ’
   33 |     EXPECT_EQ(seqan3::complement('='_sam_dna16), 'N'_sam_dna16);
      |     ^~~~~~~~~
../../../api_stability_unit/test/unit/alphabet/nucleotide/sam_dna16_test.cpp:33:50: error: unable to find character literal operator ‘operator""_sam_dna16’ with ‘char’ argument
   33 |     EXPECT_EQ(seqan3::complement('='_sam_dna16), 'N'_sam_dna16);
      |                                                  ^~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:77:52: note: in definition of macro ‘GTEST_ASSERT_’
   77 |   if (const ::testing::AssertionResult gtest_ar = (expression)) \
      |                                                    ^~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:164:3: note: in expansion of macro ‘GTEST_PRED_FORMAT2_’
```
</details>
<details><summary>Error 6: <code>error: ‘phred68legacy’ is not a member of ‘seqan3’</code></summary>

```text
[ 38%] Building CXX object alphabet/quality/CMakeFiles/quality_conversion_integration_test.dir/quality_conversion_integration_test.cpp.o
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:22:94: error: ‘phred68legacy’ is not a member of ‘seqan3’
   22 | using quality_conversion_types = seqan3::type_list<seqan3::phred42, seqan3::phred63, seqan3::phred68legacy>;
      |                                                                                              ^~~~~~~~~~~~~
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:22:107: error: template argument 3 is invalid
   22 | using quality_conversion_types = seqan3::type_list<seqan3::phred42, seqan3::phred63, seqan3::phred68legacy>;
      |                                                                                                           ^
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:22:42: error: ‘<expression error>’ in namespace ‘seqan3’ does not name a type
   22 | using quality_conversion_types = seqan3::type_list<seqan3::phred42, seqan3::phred63, seqan3::phred68legacy>;
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:23:86: error: ‘quality_conversion_types’ was not declared in this scope; did you mean ‘quality_conversion’?
   23 | using quality_conversion_gtest_types = seqan3::detail::transfer_template_args_onto_t<quality_conversion_types,
      |                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                      quality_conversion
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:24:102: error: template argument 1 is invalid
   24 |                                                                                      ::testing::Types>;
      |                                                                                                      ^
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:23:56: error: ‘<expression error>’ in namespace ‘seqan3::detail’ does not name a type
   23 | using quality_conversion_gtest_types = seqan3::detail::transfer_template_args_onto_t<quality_conversion_types,
      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   24 |                                                                                      ::testing::Types>;
      |                                                                                      ~~~~~~~~~~~~~~~~~
In file included from ../../vendor/googletest/googletest/include/gtest/gtest.h:71,
                 from ../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:13:
../../../api_stability_unit/test/unit/alphabet/quality/quality_conversion_integration_test.cpp:26:38: error: ‘quality_conversion_gtest_types’ was not declared in this scope
   26 | TYPED_TEST_SUITE(quality_conversion, quality_conversion_gtest_types, );
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest-typed-test.h:192:41: note: in definition of macro ‘TYPED_TEST_SUITE’
  192 |   typedef ::testing::internal::TypeList<Types>::type GTEST_TYPE_PARAMS_( \
      |                                         ^~~~~
```
</details>
<details><summary>Error 7: <code>error: ‘operator""_phred68legacy’ has not been declared in ‘seqan3’</code></summary>

```text
[ 39%] Building CXX object alphabet/quality/CMakeFiles/phred68legacy_test.dir/phred68legacy_test.cpp.o
../../../api_stability_unit/test/unit/alphabet/quality/phred68legacy_test.cpp:18:23: error: ‘operator""_phred68legacy’ has not been declared in ‘seqan3’
   18 | using seqan3::operator""_phred68legacy;
      |                       ^~~~~~~~~~~~~~~~
In file included from ../../vendor/googletest/googletest/include/gtest/gtest.h:375,
                 from ../../../api_stability_unit/test/unit/alphabet/quality/phred68legacy_test.cpp:8:
../../../api_stability_unit/test/unit/alphabet/quality/phred68legacy_test.cpp: In member function ‘virtual void phred68legacy_char_literal_Test::TestBody()’:
../../../api_stability_unit/test/unit/alphabet/quality/phred68legacy_test.cpp:28:31: error: unable to find character literal operator ‘operator""_phred68legacy’ with ‘char’ argument
   28 |     EXPECT_EQ(seqan3::to_char(';'_phred68legacy), ';');
      |                               ^~~~~~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:77:52: note: in definition of macro ‘GTEST_ASSERT_’
   77 |   if (const ::testing::AssertionResult gtest_ar = (expression)) \
      |                                                    ^~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:164:3: note: in expansion of macro ‘GTEST_PRED_FORMAT2_’
  164 |   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_NONFATAL_FAILURE_)
      |   ^~~~~~~~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest.h:2028:3: note: in expansion of macro ‘EXPECT_PRED_FORMAT2’
 2028 |   EXPECT_PRED_FORMAT2(::testing::internal::EqHelper::Compare, val1, val2)
      |   ^~~~~~~~~~~~~~~~~~~
../../../api_stability_unit/test/unit/alphabet/quality/phred68legacy_test.cpp:28:5: note: in expansion of macro ‘EXPECT_EQ’
   28 |     EXPECT_EQ(seqan3::to_char(';'_phred68legacy), ';');
      |     ^~~~~~~~~
../../../api_stability_unit/test/unit/alphabet/quality/phred68legacy_test.cpp:29:31: error: unable to find character literal operator ‘operator""_phred68legacy’ with ‘char’ argument
   29 |     EXPECT_EQ(seqan3::to_char('<'_phred68legacy), '<');
      |                               ^~~~~~~~~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:77:52: note: in definition of macro ‘GTEST_ASSERT_’
   77 |   if (const ::testing::AssertionResult gtest_ar = (expression)) \
      |                                                    ^~~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest_pred_impl.h:164:3: note: in expansion of macro ‘GTEST_PRED_FORMAT2_’
  164 |   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_NONFATAL_FAILURE_)
```
</details>
<details><summary>Error 8: <code>error: no match for ‘operator|’</code></summary>

```text
[ 86%] Building CXX object range/views/CMakeFiles/view_slice_test.dir/view_slice_test.cpp.o
In file included from ../../vendor/googletest/googletest/include/gtest/gtest.h:62,
                 from ../../../api_stability_unit/test/unit/range/views/view_slice_test.cpp:14:
../../../api_stability_unit/test/unit/range/views/view_slice_test.cpp: In member function ‘virtual void view_slice_underlying_is_shorter_Test::TestBody()’:
../../../api_stability_unit/test/unit/range/views/view_slice_test.cpp:128:25: error: no match for ‘operator|’ (operand types are ‘std::ranges::take_view<std::ranges::drop_view<seqan3::detail::single_pass_input_view<std::ranges::ref_view<std::__cxx11::basic_string<char> > > > >’ and ‘<unresolved overloaded function type>’)
  125 |     EXPECT_NO_THROW(( v = vec
      |                           ~~~
  126 |                         | seqan3::views::single_pass_input
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  127 |                         | seqan3::views::slice(1, 4)
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         |
      |                         std::ranges::take_view<std::ranges::drop_view<seqan3::detail::single_pass_input_view<std::ranges::ref_view<std::__cxx11::basic_string<char> > > > >
  128 |                         | seqan3::views::to<std::string> ));
      |                         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                          |
      |                                          <unresolved overloaded function type>
../../vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:1258:44: note: in definition of macro ‘GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_’
 1258 |   if (::testing::internal::AlwaysTrue()) { statement; }
      |                                            ^~~~~~~~~
../../vendor/googletest/googletest/include/gtest/gtest.h:1955:3: note: in expansion of macro ‘GTEST_TEST_NO_THROW_’
 1955 |   GTEST_TEST_NO_THROW_(statement, GTEST_NONFATAL_FAILURE_)
      |   ^~~~~~~~~~~~~~~~~~~~
../../../api_stability_unit/test/unit/range/views/view_slice_test.cpp:125:5: note: in expansion of macro ‘EXPECT_NO_THROW’
  125 |     EXPECT_NO_THROW(( v = vec
      |     ^~~~~~~~~~~~~~~
In file included from ../../../../../../seqan3/include/seqan3/range/views/to.hpp:15,
                 from ../../../api_stability_unit/test/unit/range/views/view_slice_test.cpp:22:
../../../../../../seqan3/submodules/range-v3/include/range/v3/range/conversion.hpp:59:25: note: candidate: ‘template<class MetaFn, class Pipeable> concepts::return_t<ranges::detail::to_container::closure<MetaFn, ranges::composed<Pipeable, ranges::detail::to_container::fn<MetaFn> > >, typename std::enable_if<(is_pipeable_v<Pipeable> && concepts::detail::CPP_true_fn(concepts::detail::Nil{})), void>::type> ranges::detail::operator|(ranges::detail::to_container::closure<MetaFn, ranges::detail::to_container::fn<MetaFn> > (*)(ranges::detail::to_container), Pipeable)’
   59 |             friend auto operator|(closure<MetaFn, fn<MetaFn>> (*)(to_container),
```
</details>
<details><summary>Error 9: <code>error: ‘persist’ is not a member of ‘seqan3::views’</code></summary>

```text
[ 93%] Building CXX object search/fm_index_cursor/CMakeFiles/bi_fm_index_cursor_collection_test.dir/bi_fm_index_cursor_collection_test.cpp.o
../../../api_stability_unit/test/unit/search/fm_index_cursor/bi_fm_index_cursor_collection_test.cpp:40:87: error: ‘persist’ is not a member of ‘seqan3::views’
   40 |                                                  | seqan3::views::deep{seqan3::views::persist}
      |                                                                                       ^~~~~~~
make[5]: *** [search/fm_index_cursor/CMakeFiles/bi_fm_index_cursor_collection_test.dir/build.make:63: search/fm_index_cursor/CMakeFiles/bi_fm_index_cursor_collection_test.dir/bi_fm_index_cursor_collection_test.cpp.o] Error 1
make[5]: Target 'search/fm_index_cursor/CMakeFiles/bi_fm_index_cursor_collection_test.dir/build' not remade because of errors.
make[4]: *** [CMakeFiles/Makefile2:11356: search/fm_index_cursor/CMakeFiles/bi_fm_index_cursor_collection_test.dir/all] Error 2
Scanning dependencies of target shape_test
```
</details>
<details><summary>Error 10: <code>error: ‘isnan’ is not a member of ‘std’</code></summary>

```text
[ 95%] Building CXX object std/CMakeFiles/charconv_float_test.dir/charconv_float_test.cpp.o
In file included from ../vendor/googletest/googletest/include/gtest/gtest.h:62,
                 from ../../api_stability_unit/test/unit/std/charconv_float_test.cpp:8:
../../api_stability_unit/test/unit/std/charconv_float_test.cpp: In member function ‘virtual void from_char_real_test_nan_value_Test<gtest_TypeParam_>::TestBody()’:
../../api_stability_unit/test/unit/std/charconv_float_test.cpp:218:26: error: ‘isnan’ is not a member of ‘std’
  218 |         EXPECT_TRUE(std::isnan(val));
      |                          ^~~~~
../vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:1325:34: note: in definition of macro ‘GTEST_TEST_BOOLEAN_’
 1325 |       ::testing::AssertionResult(expression)) \
      |                                  ^~~~~~~~~~
../../api_stability_unit/test/unit/std/charconv_float_test.cpp:218:9: note: in expansion of macro ‘EXPECT_TRUE’
  218 |         EXPECT_TRUE(std::isnan(val));
      |         ^~~~~~~~~~~
../../api_stability_unit/test/unit/std/charconv_float_test.cpp:227:26: error: ‘isnan’ is not a member of ‘std’
  227 |         EXPECT_TRUE(std::isnan(val));
      |                          ^~~~~
../vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:1325:34: note: in definition of macro ‘GTEST_TEST_BOOLEAN_’
 1325 |       ::testing::AssertionResult(expression)) \
      |                                  ^~~~~~~~~~
../../api_stability_unit/test/unit/std/charconv_float_test.cpp:227:9: note: in expansion of macro ‘EXPECT_TRUE’
  227 |         EXPECT_TRUE(std::isnan(val));
      |         ^~~~~~~~~~~
../../api_stability_unit/test/unit/std/charconv_float_test.cpp:236:26: error: ‘isnan’ is not a member of ‘std’
  236 |         EXPECT_TRUE(std::isnan(val));
      |                          ^~~~~
../vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:1325:34: note: in definition of macro ‘GTEST_TEST_BOOLEAN_’
 1325 |       ::testing::AssertionResult(expression)) \
      |                                  ^~~~~~~~~~
../../api_stability_unit/test/unit/std/charconv_float_test.cpp:236:9: note: in expansion of macro ‘EXPECT_TRUE’
  236 |         EXPECT_TRUE(std::isnan(val));
```
</details>

Hmm, I get locally also this error:

https://github.com/seqan/seqan3/blob/370673ffd20d790f759aed15771072436a422ec3/test/unit/search/dream_index/interleaved_bloom_filter_test.cpp#L188

```c++
/seqan3-build/api_stability-gcc-9-distcc/api_stability_unit-prefix/src/api_stability_unit/test/unit/search/dream_index/interleaved_bloom_filter_test.cpp:188:1120:   required from here
vendor/googletest/googletest/include/gtest/gtest.h:1527:11: error: no match for ‘operator==’ (operand types are ‘const seqan3::interleaved_bloom_filter<seqan3::uncompressed>::membership_agent::binning_bitvector’ and ‘const sdsl::int_vector<1>’)
```

--- a/include/seqan3/alphabet/nucleotide/dna16sam.hpp
+++ b/include/seqan3/alphabet/nucleotide/dna16sam.hpp
@@ -217,3 +217,24 @@ constexpr std::array<dna16sam, dna16sam::alphabet_size> dna16sam::complement_tab
 };
 
 } // namespace seqan3
+
+namespace seqan3
+{
+//!\deprecated Please use seqan3::dna16sam instead.
+using sam_dna16 SEQAN3_DEPRECATED_310 = seqan3::dna16sam;
+
+//!\deprecated Please use seqan3::dna16sam_vector instead.
+using sam_dna16_vector SEQAN3_DEPRECATED_310 = dna16sam_vector;
+
+//!\deprecated Please use seqan3::operator""_dna16sam instead.
+SEQAN3_DEPRECATED_310 constexpr dna16sam operator""_sam_dna16(char const c) noexcept
+{
+    return seqan3::operator""_dna16sam(c);
+}
+
+//!\deprecated Please use seqan3::operator""_dna16sam instead.
+SEQAN3_DEPRECATED_310 inline dna16sam_vector operator""_sam_dna16(char const * s, size_t n)
+{
+    return seqan3::operator""_dna16sam(s, n);
+}
+} // namespace seqan3

--- a/include/seqan3/alphabet/nucleotide/sam_dna16.hpp
+++ b/include/seqan3/alphabet/nucleotide/sam_dna16.hpp
@@ -15,11 +15,5 @@
 
 #include <seqan3/alphabet/nucleotide/dna16sam.hpp>
 
-namespace seqan3
-{
-//!\deprecated Please use seqan3::dna16sam instead.
-using sam_dna16 SEQAN3_DEPRECATED_310 = seqan3::dna16sam;
-} // namespace seqan3
-
 SEQAN3_DEPRECATED_HEADER(
    "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/alphabet/nucleotide/dna16sam.hpp> instead.")

--- a/include/seqan3/alphabet/quality/phred68legacy.hpp
+++ b/include/seqan3/alphabet/quality/phred68legacy.hpp
@@ -16,11 +16,5 @@
 
 #include <seqan3/alphabet/quality/phred68solexa.hpp>
 
-namespace seqan3
-{
-//!\deprecated Please use seqan3::phred68solexa instead.
-using phred68legacy SEQAN3_DEPRECATED_310 = seqan3::phred68solexa;
-} // namespace seqan3
-
 SEQAN3_DEPRECATED_HEADER(
    "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/alphabet/quality/phred68solexa.hpp> instead.")

--- a/include/seqan3/alphabet/quality/phred68solexa.hpp
+++ b/include/seqan3/alphabet/quality/phred68solexa.hpp
@@ -122,3 +122,21 @@ inline std::vector<phred68solexa> operator""_phred68solexa(char const * s, std::
 //!\}
 
 } // namespace seqan3
+
+namespace seqan3
+{
+//!\deprecated Please use seqan3::phred68solexa instead.
+using phred68legacy SEQAN3_DEPRECATED_310 = seqan3::phred68solexa;
+
+//!\deprecated Please use seqan3::operator""_phred68solexa instead.
+SEQAN3_DEPRECATED_310 constexpr phred68solexa operator""_phred68legacy(char const c) noexcept
+{
+    return seqan3::operator""_phred68solexa(c);
+}
+
+//!\deprecated Please use seqan3::operator""_phred68solexa instead.
+SEQAN3_DEPRECATED_310 inline std::vector<phred68solexa> operator""_phred68legacy(char const * s, size_t n)
+{
+    return seqan3::operator""_phred68solexa(s, n);
+}
+} // namespace seqan3

--- a/include/seqan3/search/dream_index/interleaved_bloom_filter.hpp
+++ b/include/seqan3/search/dream_index/interleaved_bloom_filter.hpp
@@ -592,7 +592,9 @@ public:
      * \private
      * \param ibf The seqan3::interleaved_bloom_filter.
      */
-    membership_agent(ibf_t const & ibf) : ibf_ptr(std::addressof(ibf)), result_buffer(ibf.bin_count()) {}
+    explicit membership_agent(ibf_t const & ibf) :
+        ibf_ptr(std::addressof(ibf)), result_buffer(ibf.bin_count())
+    {}
     //!\}
 
     //!\brief Stores the result of bulk_contains().
@@ -679,7 +681,9 @@ public:
     ~binning_bitvector() = default; //!< Defaulted.
 
     //!\brief Construct with given size.
-    binning_bitvector(size_t const size) : data(size) {}
+    explicit binning_bitvector(size_t const size) :
+        data(size)
+    {}
     //!\}
 
     //!\brief Returns the number of elements.
@@ -916,7 +920,7 @@ public:
      * \private
      * \param ibf The seqan3::interleaved_bloom_filter.
      */
-    counting_agent_type(ibf_t const & ibf) :
+    explicit counting_agent_type(ibf_t const & ibf) :
         ibf_ptr(std::addressof(ibf)), membership_agent(ibf), result_buffer(ibf.bin_count())
     {}
     //!\}

--- a/include/seqan3/search/dream_index/interleaved_bloom_filter.hpp
+++ b/include/seqan3/search/dream_index/interleaved_bloom_filter.hpp
@@ -734,6 +734,36 @@ public:
     {
         return !(lhs == rhs);
     }
+
+#ifdef SEQAN3_DEPRECATED_310
+    //!\brief Test for equality.
+    //!\deprecated Use binning_bitvector.raw_data() == rhs
+    SEQAN3_DEPRECATED_310 friend bool operator==(binning_bitvector const & lhs, sdsl::bit_vector const & rhs) noexcept
+    {
+        return lhs.data == rhs;
+    }
+
+    //!\brief Test for equality.
+    //!\deprecated Use lhs == binning_bitvector.raw_data()
+    SEQAN3_DEPRECATED_310 friend bool operator==(sdsl::bit_vector const & lhs, binning_bitvector const & rhs) noexcept
+    {
+        return lhs == rhs.data;
+    }
+
+    //!\brief Test for inequality.
+    //!\deprecated Use binning_bitvector.raw_data() != rhs
+    SEQAN3_DEPRECATED_310 friend bool operator!=(binning_bitvector const & lhs, sdsl::bit_vector const & rhs) noexcept
+    {
+        return !(lhs.data == rhs);
+    }
+
+    //!\brief Test for inequality.
+    //!\deprecated Use lhs != binning_bitvector.raw_data()
+    SEQAN3_DEPRECATED_310 friend bool operator!=(sdsl::bit_vector const & lhs, binning_bitvector const & rhs) noexcept
+    {
+        return !(lhs == rhs.data);
+    }
+#endif // SEQAN3_DEPRECATED_310
     //!\}
 
     /*!\name Access

--- a/test/api_stability/3.0.2/0023-NOAPI-INCLUDE-add-missing-includes-seqan3-views-pers.patch
+++ b/test/api_stability/3.0.2/0023-NOAPI-INCLUDE-add-missing-includes-seqan3-views-pers.patch
@@ -1,0 +1,38 @@
+From 12b7805b266a7477492657756fa77dda9b8f5fe5 Mon Sep 17 00:00:00 2001
+From: marehr <marehr-github@marehr.dialup.fu-berlin.de>
+Date: Sun, 25 Apr 2021 18:12:55 +0200
+Subject: [PATCH 23/24] [NOAPI] [INCLUDE] add missing includes
+ seqan3::views::persist
+
+---
+ .../pairwise_alignment/pairwise_alignment_solution_6.cpp         | 1 +
+ .../fm_index_cursor/bi_fm_index_cursor_collection_test.cpp       | 1 +
+ 2 files changed, 2 insertions(+)
+
+diff --git a/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp b/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp
+index 1ed35c8cd..32b6077d2 100644
+--- a/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp
++++ b/doc/tutorial/pairwise_alignment/pairwise_alignment_solution_6.cpp
+@@ -6,6 +6,7 @@
+ #include <seqan3/alphabet/nucleotide/dna4.hpp>
+ #include <seqan3/core/debug_stream.hpp>
+ #include <seqan3/range/views/pairwise_combine.hpp>
++#include <seqan3/range/views/persist.hpp>
+ #include <seqan3/std/ranges>
+ 
+ using seqan3::operator""_dna4;
+diff --git a/test/unit/search/fm_index_cursor/bi_fm_index_cursor_collection_test.cpp b/test/unit/search/fm_index_cursor/bi_fm_index_cursor_collection_test.cpp
+index 48be041bc..5d055adb9 100644
+--- a/test/unit/search/fm_index_cursor/bi_fm_index_cursor_collection_test.cpp
++++ b/test/unit/search/fm_index_cursor/bi_fm_index_cursor_collection_test.cpp
+@@ -9,6 +9,7 @@
+ 
+ #include <seqan3/alphabet/nucleotide/dna4.hpp>
+ #include <seqan3/alphabet/nucleotide/dna5.hpp>
++#include <seqan3/range/views/persist.hpp>
+ #include <seqan3/range/views/slice.hpp>
+ #include <seqan3/range/views/char_to.hpp>
+ #include <seqan3/range/views/to.hpp>
+-- 
+2.31.1
+

--- a/test/api_stability/3.0.2/0024-NOAPI-INCLUDE-add-missing-includes-std-isnan.patch
+++ b/test/api_stability/3.0.2/0024-NOAPI-INCLUDE-add-missing-includes-std-isnan.patch
@@ -1,0 +1,24 @@
+From ccf8a01d614f20c114d384003092a60366056be8 Mon Sep 17 00:00:00 2001
+From: marehr <marehr-github@marehr.dialup.fu-berlin.de>
+Date: Sun, 25 Apr 2021 18:13:15 +0200
+Subject: [PATCH 24/24] [NOAPI] [INCLUDE] add missing includes std::isnan
+
+---
+ test/unit/std/charconv_float_test.cpp | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/test/unit/std/charconv_float_test.cpp b/test/unit/std/charconv_float_test.cpp
+index 123f2e3d8..654d2a8c2 100644
+--- a/test/unit/std/charconv_float_test.cpp
++++ b/test/unit/std/charconv_float_test.cpp
+@@ -7,6 +7,7 @@
+ 
+ #include <gtest/gtest.h>
+ 
++#include <cmath>
+ #include <iostream>
+ #include <limits>
+ 
+-- 
+2.31.1
+

